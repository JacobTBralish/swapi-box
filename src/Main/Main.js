import React from 'react';
import CardContainer from '../CardContainer/CardContainer'
import Welcome from '../Welcome/Welcome'
import { getPeople, getPlanets, getVehicles } from '../api'
import { Switch, Route } from 'react-router-dom'
import './Main.css'

const Main = ({ scroll }) => {
  return( 
    <div className="container">
      <Switch>
        <Route exact path='/' render={() => (
          <Welcome scroll={scroll} /> 
        )} />
        <Route path='/people' render={() => (
          <CardContainer name='people' fetch={getPeople} />
        )} />
        <Route path='/planets' render={() => (
          <CardContainer name='planets' fetch={getPlanets} />
        )} />
       <Route path='/vehicles' render={() => (
          <CardContainer name='vehicles' fetch={getVehicles} />
        )} />
       <Route path='/favorites' render={() => (
        <CardContainer name='favorites' />
        )} />
      </Switch>
    </div>
  )
}


export default Main;