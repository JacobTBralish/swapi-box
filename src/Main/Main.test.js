/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import Main from './Main';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Main Component', () => {
  test('it should render without crashing', () => {
    const renderedComp = shallow(<Main />)

    expect(renderedComp).toMatchSnapshot()
  })
})