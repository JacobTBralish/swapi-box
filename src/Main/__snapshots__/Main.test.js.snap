// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component it should render without crashing 1`] = `
<div
  className="container"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/people"
      render={[Function]}
    />
    <Route
      path="/planets"
      render={[Function]}
    />
    <Route
      path="/vehicles"
      render={[Function]}
    />
    <Route
      path="/favorites"
      render={[Function]}
    />
  </Switch>
</div>
`;
